crates/floresta-wire/src/p2p_wire/sync_node.rs

```diff
@@ -158,13 +158,16 @@ where
             if block.udata.is_none() {
                 error!("Block without proof received from peer {}", peer);
                 self.send_to_peer(peer, NodeRequest::Shutdown).await?;
-                self.send_to_random_peer(
-                    NodeRequest::GetBlock((vec![block.block.block_hash()], true)),
-                    ServiceFlags::UTREEXO,
-                )
-                .await?;
-                self.inflight
-                    .insert(InflightRequests::Blocks(next_block), (peer, Instant::now()));
+                let next_peer = self
+                    .send_to_random_peer(
+                        NodeRequest::GetBlock((vec![block.block.block_hash()], true)),
+                        ServiceFlags::UTREEXO,
+                    )
+                    .await?;
+                self.inflight.insert(
+                    InflightRequests::Blocks(next_block),
+                    (next_peer, Instant::now()),
+                );
```

Comment:
Previously, we were sending requests to the random peer correctly, but creating inflight requests to that banned peer only. Now it is changed to the next random peer we sending the request to.